### PR TITLE
Add exclude and include config for filter log processor

### PR DESCRIPTION
**Pull request by chaitanyaphalak**
**Description:** Changing the configs for logs filter processor to be consistent with the metrics filter processor and add ability to include logs based on resource attributes in addition to excluding logs based on resource attributes for strict matching. 

**Testing:** Unit tests updated for the the new configs

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]
LGTM. Why still in draft?

File: processor/filterprocessor/filter_processor_logs_test.go [hunk 1]
```diff
@@ -73,23 +74,45 @@ var (
...
+			},
+		},
+		{
+			name:   "includeAllWithMissingResourceAttributes",
+			inc:    &LogMatchProperties{LogMatchType: Strict, ResourceAttributes: []filterconfig.Attribute{{Key: "attr1", Value: "attr1/val2"}}},
+			inLogs: testResourceLogs(inLogForTwoResource),
+			outLN: [][]string{
+				{"log3", "log4"},
+			},
+		},
```
Comment (dmitryax) [inline processor/filterprocessor/filter_processor_logs_test.go:98]: Let's add a test case with both include and exclude

**Comment by chaitanyaphalak** [comment]
@dmitryax wanted to test with our helm chart before the merge and change anything if needed :)

**Comment by chaitanyaphalak** [comment]
@dmitryax added test cases for include and exclude.

File: receiver/dockerstatsreceiver/docker_test.go [hunk 2]
```diff
@@ -12,7 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !windows
 // +build !windows
+
```
Comment (dmitryax) [inline receiver/dockerstatsreceiver/docker_test.go:17]: Please remove these as unrelated changes. Feel free to submit another PR to add the new style build comments

**Review by dmitryax — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]
We got a release cut with the previous config interface. Please add a line to Changelog as a breaking change.

**Review by pmalek-sumo — APPROVED** [review]
Looking forward for this change. 👍 

One comment/question though: shouldn't we also update the README?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,12 +2,20 @@
 
 ## Unreleased
 
+## 🛑 Breaking changes 🛑
+
+- `filter` processor: The configs for `logs` filter processor have been changed to be consistent with the `metrics` filter processor. (#4895)
+
+## 💡 Enhancements 💡
+
+- `tailsampling` processor: Add new policy `probabilistic` (#3876)
```
[the remark this answers is not in the document]
Comment (chaitanyaphalak) [reply CHANGELOG.md]: Looks like a bad merge

**Review by dmitryax — APPROVED** [review]

**Comment by pmalek-sumo** [comment]
I've been testing this PR in my build and one thing that surprised me (which might be obvious to others) is that the filters seem to be logically `AND`ed so I cannot use the following filter:

```yaml
 ...
        processors:
          groupbyattrs/systemd:
            keys:
              - _SYSTEMD_UNIT
          filter/systemd:
            logs:
              include:
                match_type: strict
                resource_attributes:
                  - key: _SYSTEMD_UNIT
                    value: dbus.service
                  - key: _SYSTEMD_UNIT
                    value: docker.service
...
```

because it never matches both of those (I believe this can be done by tagging systemd related log entries with a specific tag but haven't gotten to this point yet).

Any suggestions on how to approach it (the only other way apart from what I suggested above with tagging is to explicitly state all possible values in their own respective processors but this gets quickly out of hand).

Regexp filters would obviously help here but I'm not sure what are the plans with respect to that.

**Comment by dmitryax** [comment]
Hey @pmalek-sumo , thank you for raising this!

This is indeed a real issue in resource attributes matching applicable to both logs and [metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/297da44e54cca051b6db5038edacba0e0a8133a6/processor/filterprocessor/filter_processor.go#L169). It get's even more confusing in the `exclude` section, it just doesn't work if there are 2 or more different entries in `resource_attributes`.

Do you have a chance to submit an issue for that? We need to fix the matching logic uniformly for metrics and traces which will be another "breaking" change.

This PR is doesn't effect the functionality, but still keeps it a broken state.

**Comment by dmitryax** [comment]
@pmalek-sumo I've submitted an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5234

**Pull request merged**